src/layouts/dodge.js

```diff
@@ -39,8 +39,9 @@ export function dodgeY(dodgeOptions = {}, options = {}) {
 }
 
 function dodge(y, x, anchor, padding, options) {
-  const [, r] = maybeNumberChannel(options.r, 3);
-  return layout(options, (I, scales, {[x]: X, r: R}, dimensions) => {
+  const [, r0] = maybeNumberChannel(options.r, 3);
+  return layout(options, function(I, scales, {[x]: X, r: R}, dimensions) {
+    const r = this.r == null ? r0 : this.r;
```

Comment:
In cases of option “inheritance” or defaults, we typically check against undefined rather than null. Also, we should try to avoid computing values that aren’t needed. The precedence order here is:

* The `r` channel (`R`)
* The `r` mark property (`this.r`)
* The `r` option (`options.r`)

We may not even need maybeNumberChannel here, since if `options.r` were interpreted as a channel, then `R` would be defined, so we can use it.

```suggestion
  return layout(options, function(I, scales, {[x]: X, r: R}, dimensions) {
    const r = R ? undefined : this.r !== undefined ? +this.r : options.r !== undefined ? +options.r : 3;
```

(I kinda want to use the null coalescing operator `??` here, but I think the correct interpretation of null here is zero rather than undefined, since null represents a defined value. I kinda wish they made it consistent with default arguments.)